geckocamera-codec.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2021 Open Mobile Platform LLC.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <dlfcn.h>
+#include <map>
+#include <filesystem>
+
+#include "geckocamera-codec.h"
+
+#define LOG_TOPIC "codec"
+#include "geckocamera-plugins.h"
+#include "geckocamera-utils.h"
+
+namespace gecko {
+namespace codec {
+
+using namespace std;
+using namespace gecko::camera;
+
+class RootCodecManager : public CodecManager
+{
+public:
+    RootCodecManager() {};
+    ~RootCodecManager() {};
+
+    bool init();
+
+    bool videoEncoderAvailable(CodecType codecType);
+    bool videoDecoderAvailable(CodecType codecType);
+
+    bool createVideoEncoder(CodecType codecType, shared_ptr<VideoEncoder> &encoder);
+    bool createVideoDecoder(CodecType codecType, shared_ptr<VideoDecoder> &decoder);
+
+private:
+    shared_ptr<CodecManager> loadPlugin(Plugin &plugin);
+
+    bool initialized = false;
+    map<const string, shared_ptr<CodecManager>> plugins;
+    map<const string, shared_ptr<CodecManager>> mimeTypeMap;
```

Comment:
Don't recall how other classes have been written but something like mMimeTypeMap for each member variable would ease reading the code.

Reply:
I'll stick to `m_` prefix as gmp-droid does, if you don't mind. I fixed notation for the class members for this PR.

Reply:
That's fine. Thank you.